testing/kfctl/kf_is_ready_checklists/kfctl_existing_arrikto_checklist.yaml

```diff
@@ -0,0 +1,153 @@
+# Deployments in namespace kubeflow
```

Comment:
Instead of making this Arrikto specific, should we have a file that is components common to all deployments and then a separate file for anything that is config specific?

Reply:
I was thinking of making more changes here, to generate this list from KfDef Spec used in testing. The approach would be: look into the applications used in KfDef spec, search inside of the generated manifests folder for matching deployments/statefulset etc. I can take a look at this once #4148, this PR and #4187 is merged.
Does that sound reasonable.

Reply:
@nrchakradhar  would be great if we could automate this.
@jlewi the current python test is GCP-specific, that's why I had to find a way to specify what I wanted.
Ideally, the E2E test will discover what is needed.
Otherwise, we could define it in checklist-like files, which can be used by any config.

Reply:
@yanniszark see TODO here
https://github.com/kubeflow/kubeflow/blob/e38c6f3234b301f33226706c285aafb9bd32d357/testing/kfctl/kf_is_ready_test.py#L33

There are a set of applications that should be installed across all deployments (e.g. TFJob, PyTorch,, Pipelines etc....).

Think of this as a compliance tests. These are the applications that should be installed on all KF compliant clusters.

So that list whether specified in a YAML file or in python should be the same for all applications.

Then there should be a list of applications that should only be installed in certain settings; e.g. on GCP with IAP.

So the tests that apply to all components should be written once and reused across all components.

So we should create a list of all the K8s resources common to all deployments.
Then based on either the KFDef spec or command line arguments we should add additional expectations.

So to make the test non GCP specific we should conditionally enable the set of K8s resources to look for GCP resources based on whether the GCP plugin spec is provided in the KFDef file.

Reply:
Thanks @yanniszark for updating kfctl_go_test.py.
Can we remove this file now?